src/main/kotlin/org/koitharu/kotatsu/parsers/site/en/ComicExtra.kt

```diff
@@ -0,0 +1,151 @@
+		page: Int,
+		query: String?,
+		tags: Set<MangaTag>?,
+		sortOrder: SortOrder,
+	): List<Manga> {
+		val tag = tags.oneOrThrowIfMany()
+		val url = buildString {
+			append("https://$domain/")
+			if (!tags.isNullOrEmpty()) {
+				append(tag?.key.orEmpty())
+				if (page > 1) {
+					append(page)
+				}
+			} else if (!query.isNullOrEmpty()) {
+				append("comic-search?key=")
+				append(query.urlEncoded())
+				if (page > 1) {
+					append("&page=")
+					append(page)
+				}
+			} else {
+				when (sortOrder) {
+					SortOrder.POPULARITY -> append("popular-comic/")
+					SortOrder.UPDATED -> append("new-comic/")
+					SortOrder.NEWEST -> append("recent-comic/")
+					else -> append("new-comic/")
+				}
+				if (page > 1) {
+					append(page)
+				}
+			}
+		}
+
+		val doc = webClient.httpGet(url).parseHtml()
+
+		return doc.select("div.movie-list-index div.cartoon-box").map { div ->
+			val href = div.selectFirstOrThrow("a").attrAsAbsoluteUrl("href")
+			Manga(
+				id = generateUid(href),
+				title = div.selectFirstOrThrow("h3").text(),
+				altTitle = null,
+				url = href,
+				publicUrl = href.toAbsoluteUrl(domain),
+				rating = RATING_UNKNOWN,
+				isNsfw = false,
+				coverUrl = div.selectFirstOrThrow("img").attrAsAbsoluteUrl("src"),
+				tags = emptySet(),
+				state = when (div.selectFirstOrThrow(".detail:contains(Stasus: )").text()) {
+					"Stasus: Ongoing" -> MangaState.ONGOING
+					"Stasus: Completed" -> MangaState.FINISHED
+					else -> null
+				},
+				author = null,
+				source = source,
+			)
+		}
+	}
+
+
+	override suspend fun getTags(): Set<MangaTag> {
+		val doc = webClient.httpGet("https://$domain/popular-comic").parseHtml()
+		return doc.select("li.tag-item a").mapNotNullToSet { a ->
+			MangaTag(
+				key = a.attr("href").substringAfterLast("/"),
+				title = a.text(),
+				source = source,
+			)
+		}
+	}
+
+	override suspend fun getDetails(manga: Manga): Manga {
+		val doc = webClient.httpGet(manga.url.toAbsoluteUrl(domain)).parseHtml()
+		return manga.copy(
+			altTitle = doc.selectFirstOrThrow("dt.movie-dt:contains(Alternate name:) + dd").text(),
+			state = when (doc.selectFirstOrThrow("dt.movie-dt:contains(Status:) + dd a").text()) {
+				"Ongoing" -> MangaState.ONGOING
+				"Completed" -> MangaState.FINISHED
+				else -> null
+			},
+			tags = doc.select("dt.movie-dt:contains(Genres:) + dd a").mapNotNullToSet { a ->
+				MangaTag(
+					key = a.attr("href").substringAfterLast("/"),
+					title = a.text(),
+					source = source,
+				)
+			},
+			author = doc.select("dt.movie-dt:contains(Author:) + dd").text(),
+			description = doc.getElementById("film-content")?.text(),
+			chapters = doc.requireElementById("list").select("tr")
+				.mapChapters(reversed = true) { i, tr ->
+
+					val a = tr.selectFirstOrThrow("a")
+					val url = a.attrAsRelativeUrl("href") + "/full"
+					val name = a.text()
+					val dateText = tr.select("td").last()?.text()
+					val dateFormat = SimpleDateFormat("MM/dd/yy", sourceLocale)
```

Comment:
There are no reason to create new `SimpleDateFormat` instance every time, it might be better to move this line out of the `mapChapters` lambda